androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/splash/SplashScreen.kt

```diff
@@ -1,78 +1,260 @@
 
 import androidx.compose.animation.AnimatedVisibility
-import androidx.compose.animation.core.tween
+import androidx.compose.animation.core.*
 import androidx.compose.animation.fadeIn
 import androidx.compose.foundation.Image
 import androidx.compose.foundation.background
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
 import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.*
+import androidx.compose.ui.geometry.Offset
 import androidx.compose.ui.graphics.Brush
-import androidx.compose.ui.graphics.Color.Companion.Transparent
+import androidx.compose.ui.graphics.Color.Companion.Black
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.input.*
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
 import co.nimblehq.avishek.phong.kmmic.android.R
+import co.nimblehq.avishek.phong.kmmic.android.ui.common.PrimaryButton
+import co.nimblehq.avishek.phong.kmmic.android.ui.common.PrimaryTextField
 import co.nimblehq.avishek.phong.kmmic.android.ui.theme.ApplicationTheme
-import co.nimblehq.avishek.phong.kmmic.android.ui.theme.Black60
 import kotlinx.coroutines.delay
 
-private const val DELAY_LOGO = 500L
-private const val DURATION_LOGO = 1000
+private const val LogoDelayInMillis = 500L
+private const val LogoDurationInMillis = 750
+private const val LoginFormRevealDurationInMillis = 700
+
+const val LoginEmailFieldContentDescription = "LoginEmailField"
+const val LoginPasswordFieldContentDescription = "LoginPasswordField"
+const val LoginButtonContentDescription = "LoginButton"
+
+private val InitialLogoOffset: Offset = Offset(0f, 0f)
+private val FinalLogoOffset = Offset(0f, -230f)
+private const val InitialLogoScale: Float = 1f
+private const val FinalLogoScale: Float = 0.83f
+private const val InitialBlurRadius: Float = 0f
+private const val FinalBlurRadius = 25f
+private const val InitialAlpha: Float = 0f
+private const val FinalAlpha: Float = 1f
+private const val InitialTopGradientAlpha: Float = 0.2f
+private const val InitialBottomGradientAlpha: Float = 0.6f
+private const val BottomGradientAlphaMultiplier: Float = 1 - InitialBottomGradientAlpha
+
+private const val ForgotTextAlpha = 0.5f
 
 @Composable
 fun SplashScreen() {
-    var shouldShowLogo by remember {
-        mutableStateOf(false)
-    }
+    var shouldShowLogo by remember { mutableStateOf(false) }
+    var email by remember { mutableStateOf("") }
+    var password by remember { mutableStateOf("") }
```

Comment:
We don't really need to place this in the splash screen. It doesn't belong to Splash. It belongs to `LoginForm`.

Reply:
Legit point. But we can keep the `onLoginClick` in `SplashScreen`, right? We'll call the VM in that function.